### ISSUE TITLE
[BUG] Miner endless-loop when water flow in the way of node

### ISSUE THREAD

**Issue by leagris** [ldtteam/minecolonies#8153]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.18

### MineColonies Version

1.0.494-ALPHA

### Structurize Version

1.0.352-ALPHA

### Related Mods and their Versions

1. Forge version: 39.0.75
2. BlockUI version: 0.0.47-ALPHA
3. Domum-Ornatum version: .0.39-ALPHA

### Current Behavior

As @MotionlessTrain suggested:
> 21:57] MotionlessTrain: It already does for the incan style. The mining nodes themselves have water channels in them

When water is flowing in the way of a new mine node being built, the miner might get stuck in an infinite loop, placing cobblestone to plug the water flow, then mining back the same cobble block in a later round, and cyclin back infinitely replacing and breaking forever.

### Expected Behavior

Another more appropriate algorithm would plug the liquid flow with solid blocks from the structure and not in the way of blocks to be mined-back.

### Reproduction Steps

1. Choose Medieval Oak mine
2. Wait for miner to deploy node in the way of water flow
3. Likely a frequent situation in 1.18 with so many watered caves.

### Logs

https://pastebin.com/Q36j6rnU

### Anything else?

Forever placing and breaking cobblestone when water flow:

![Forever placing and breaking cobblestone when water flow](https://user-images.githubusercontent.com/1111474/158696556-adfca9e1-42cd-47bb-8a86-a9113b11ab81.png)

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by sreny1** [comment]
same issue happend to me in 1.0.778-ALPHA version with wooden and stone style mines

also continues looping after sending the miner to a different level (he goes to the assigned level, but continues looping the blocks on the other level)

even fire and reasign the miner get the loop to stop

**Comment by MotionlessTrain** [comment]
This can happen with waterlogged blocks as well: Blocks like fences, and apparently pointed dripstone are seen as solid blocks by the miner, and won't be replaced by a fill block. If they are waterlogged, that water never gets cleared out and will cause this behaviour as well

**Comment by doqunbop** [comment]
I observed the same bug in my mine yesterday when my miner was digging up a piece of cobblestone and then putting it back, without any other cobblestone in the bags, at which point I manually placed blocks (dirt and cobblestone) to rid of the water. Later it got to a different spot with the same problem when I noticed the same pattern, but this time whenever the miner was placing the cobblestone back, it would place one to three blocks at a time, and given enough time was able to get past the water filled area.
So the bug might be that the miner does not detect that it's lacking cobblestone to rid of the water, only placing as much as they have before mining it back up and repeating.
